src/ex2/main.js

```diff
@@ -1,14 +1,219 @@
+        // according to font awesome buttonIcon are "check" or "trash"
+        const button1 = this.createButtonAndAddClass(
+            "complete-button",
+            "check"
+        );
+        const button2 = this.createButtonAndAddClass("trash-button", "trash");
+        return [button1, button2];
+    };
+
+    createButtonAndAddClass = (buttonClass, buttonIcon) => {
+        const newButton = document.createElement("button");
+        //i tag is for font awesome
+        newButton.innerHTML = `<i class="fas fa-${buttonIcon}" ></i>`;
+        newButton.classList.add(buttonClass);
+        return newButton;
+    };
+
+    createPokemonSprit = async (pokemonName) => {
+        const newImg = document.createElement("img");
+        newImg.src = await pokemonClient.getPokemonImgByName(pokemonName);
+        newImg.alt = `${pokemonName} sprit`;
+        newImg.style = "width:50px;height:50px;";
+        return newImg;
+    };
+
+    deleteOrCompleteTask = (event) => {
+        const item = event.target;
+        if (item.classList[0] === "trash-button") {
+            const todo = item.parentElement;
+            todo.classList.add("fall");
+            setTimeout(() => {
+                itemManager.removeTask(todo.childNodes[0].innerHTML);
+                todo.remove();
+                // this.renderTasks() //comment out because of performance.
+                // The readme file says to render again from the array after each delete,
+                // my logic delete the dom element so no need to re-render.
+            }, 500);
+        } else if (item.classList[0] === "complete-button") {
+            const todo = item.parentElement;
+            todo.classList.toggle(TASK_COMPLETED_CLASS_NAME);
+        } else if (item.classList[0] === "todo-item") {
+            confirm(item.innerHTML);
+        }
+    };
+
+    showHideFeaturesButtons = () => {
+        if (this.todos.length !== 0) {
+            this.clearAllButton.style.display = "flex";
+            this.filterOption.style.display = "flex";
+        } else {
+            this.clearAllButton.style.display = "none";
+            this.filterOption.style.display = "none";
+        }
+    };
+
+    updatePending = () => {
+        const completedTask = document.querySelectorAll(".completed");
+        const pendingTasksAmount = this.todos.length - completedTask.length;
+        if (pendingTasksAmount === 0) {
+            this.pendingCounter.innerHTML = `YAYY all your task are done, add new task...`;
+        } else {
+            this.pendingCounter.innerHTML = `You have ${pendingTasksAmount} pending tasks`;
+        }
+        this.showHideFeaturesButtons();
+    };
+
+    callback = (mutationsList) => {
+        for (const mutation of mutationsList) {
+            if (mutation.type === "childList") {
+                this.updatePending();
+            } else if (mutation.type === "attributes") {
+                this.updatePending();
+            }
```

Comment:
you can combine this if to a single one

```suggestion
            if (["childList", "attributes"].include(mutation.type)) {
                this.updatePending();
            }
```